ast/location/location.go

```diff
@@ -87,3 +93,34 @@ func (loc *Location) Compare(other *Location) int {
 	}
 	return 0
 }
+
+func (loc *Location) MarshalJSON() ([]byte, error) {
+	// structs are used here to preserve the field ordering of the original Location struct
+	var data interface{}
+	if loc.JSONOptions.MarshalOptions.IncludeLocationText {
+		data = struct {
+			File string `json:"file"`
+			Row  int    `json:"row"`
+			Col  int    `json:"col"`
+			Text []byte `json:"text"`
+		}{
+			File: loc.File,
+			Row:  loc.Row,
+			Col:  loc.Col,
+			Text: loc.Text,
+		}
+		return json.Marshal(data)
+	}
+
+	data = struct {
+		File string `json:"file"`
+		Row  int    `json:"row"`
+		Col  int    `json:"col"`
+	}{
+		File: loc.File,
+		Row:  loc.Row,
+		Col:  loc.Col,
+	}
+
+	return json.Marshal(data)
+}
```

Comment:
I _believe_ we can deduplicate this a bit... but it's finicky, I might be on the wrong track 😅 

```suggestion
func (loc *Location) MarshalJSON() ([]byte, error) {
	// structs are used here to preserve the field ordering of the original Location struct
	data := struct {
		File string `json:"file"`
		Row  int    `json:"row"`
		Col  int    `json:"col"`
		Text []byte `json:"text,omitempty"`
	}{
		File: loc.File,
		Row:  loc.Row,
		Col:  loc.Col,
	}
	if loc.JSONOptions.MarshalOptions.IncludeLocationText {
		data.Text = loc.Text
	}
	return json.Marshal(data)
}
```

Reply:
I think that this is subtly different in that if the text were to be blank, it'd now be omitted. I think that's ok though!

Reply:
Good point! I hadn't thought about that edge case. Your call, you understand the needs better than I do here :)